lib/node_modules/@stdlib/_tools/eslint/rules/tsdoc-declarations-doctest/lib/main.js

```diff
@@ -0,0 +1,498 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/* eslint-disable max-len, max-lines */
+
+'use strict';
+
+// MODULES //
+
+var vm = require( 'vm' );
+var resolve = require( 'path' ).resolve;
+var readFileSync = require( 'fs' ).readFileSync; // eslint-disable-line node/no-sync
+var logger = require( 'debug' );
+var resolveParentPath = require( '@stdlib/fs/resolve-parent-path' ).sync;
+var dirname = require( '@stdlib/utils/dirname' );
+var isNull = require( '@stdlib/assert/is-null' );
+var isNumber = require( '@stdlib/assert/is-number' );
+var contains = require( '@stdlib/assert/contains' );
+var replace = require( '@stdlib/string/replace' );
+var endsWith = require( '@stdlib/string/ends-with' );
+var trim = require( '@stdlib/string/trim' );
+var reEOL = require( '@stdlib/regexp/eol' );
+var escapeRegExpString = require( '@stdlib/utils/escape-regexp-string' );
+var objectKeys = require( '@stdlib/utils/keys' );
+var tryRequire = require( '@stdlib/utils/try-require' );
+var compareValues = require( '@stdlib/_tools/doctest/compare-values' );
+var createAnnotationValue = require( '@stdlib/_tools/doctest/create-annotation-value' );
+var createVMScope = require( './create_vm_scope.js' );
+var addPackageToScope = require( './add_package_to_scope.js' );
+var SCOPE_DEFAULTS = require( './scope_defaults.json' );
+
+
+// VARIABLES //
+
+var debug = logger( 'tsdoc-doctest' );
+var RE_TSDOC = /\/\*\*[\s\S]+?\*\//g;
+var RE_EXAMPLE = /@example\s*([\s\S]*?)(?=\n\s*@\w|\*\/|$)/g;
+var RE_ANNOTATION = /(?:\n|^)(?:var|let|const)? ?([a-zA-Z0-9._]+) ?=?[^;]*;\n\/\/ ?(returns|([A-Za-z][A-Za-z_0-9]*)? ?=>|throws) ?([\s\S]*?)(?=\n|$)/g;
+var RE_COMMENT_PREFIX = /^\s*\*\s?/gm;
+var DEFAULT_ERROR_LOC = {
+	'start': {
+		'line': 1,
+		'column': 0
+	},
+	'end': {
+		'line': 1,
+		'column': 0
+	}
+};
+var regexCache = {};
+var lineCountCache = {};
+var rule;
+
+
+// FUNCTIONS //
+
+/**
+* Gets or creates a cached regex for finding return annotations.
+*
+* @private
+* @param {string} annotationType - type of annotation (returns, throws, etc.)
+* @param {string} value - expected value
+* @returns {RegExp} compiled regular expression
+*/
+function getAnnotationRegex( annotationType, value ) {
+	var escapedValue;
+	var pattern;
+	var key;
+
+	escapedValue = escapeRegExpString( value );
+	key = annotationType + '::' + escapedValue;
+
+	if ( !regexCache[ key ] ) {
+		pattern = '// ' + annotationType + '\\s+' + escapedValue;
+		regexCache[ key ] = new RegExp( pattern );
+	}
+	return regexCache[ key ];
+}
+
+/**
+* Counts the number of lines in the given string.
+*
+* @private
+* @param {string} str - input string
+* @returns {number} number of lines
+*/
+function countLines( str ) {
+	var lines;
+
+	// Use cached result if available
+	if ( lineCountCache[ str ] ) {
+		return lineCountCache[ str ];
+	}
+	lines = str.split( reEOL() );
```

Comment:
@Planeshifter You don't need to create a new EOL regexp here. Use `.REGEXP` from the respective package.